### PR TITLE
Adds landing page for canceled move [delivers #158244741]

### PR DESCRIPTION
**Pull request by stangah**
## Description

Adds a landing page for canceled moves as per the wireframe in the linked Pivotal Tracker ticket. Some additional things:
- I couldn't live with myself if I just added another conditional to `MoveSummary.jsx`, so I refactored the thing to be more maintainable and easy to read. Incidentally the diff is hard to follow, so it might be easier to look at before/after separately
- Rather than adding another query to fetch transportation office info, I started passing it through with existing `duty_station` info. In a future PR I'll refactor existing code to take advantage of this
- Per the wireframe, an alert will show on the landing page when a move was canceled. As it is in this PR, that alert will persist indefinitely. This is the immediate plan as I understand it, and `Alert` management is going to be addressed soon (before any moves are actually canceled, ideally)

## Code Review Verification Steps

* [x] All tests pass.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/blob/master/docs/backend.md#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/blob/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.
* [ ] Any migrations/schema changes also update the diagram in docs/schema/dp3.sqs
* [ ] There are no aXe warnings for UI.
* [ ] This works in IE.
* Any new client dependencies (Google Analytics, hosted libraries, CDNs, etc) have been:
  * [ ] Communicated to @willowbl00
  * [ ] Added to the list of [network dependencies](https://github.com/transcom/mymove#client-network-dependencies)
* [x] Request review from a member of a different team.
* [ ] (TRIAL) Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/158244741) for this change

## Screenshots

screenshot 2018-06-14 15 28 08

### CODE REVIEW

File: src/scenes/Landing/MoveSummary.jsx [hunk 1]
```diff
@@ -12,342 +12,402 @@ import ppmSubmitted from './images/ppm-submitted.png';
...
+              <CanceledMoveSummary
+                ppm={ppm}
+                orders={orders}
+                profile={profile}
+                move={move}
+                entitlement={entitlement}
+                reviewProfile={reviewProfile}
+              />
+            ),
+          }[status]
```
Comment (jim) [inline src/scenes/Landing/MoveSummary.jsx]: I find this construction a little hard to read. It might be worth trying to pick the component to display before the `return` statement on line 332 and then just embedding that value in the JSX literal.

What do other folks think?

File: src/scenes/Moves/ducks.js [hunk 2]
```diff
@@ -88,10 +89,10 @@ function reshapeMove(move) {
 export function moveReducer(state = initialState, action) {
   switch (action.type) {
     case GET_LOGGED_IN_USER.success:
+      const moves = get(action.payload, 'service_member.orders.0.moves', []);
       return Object.assign({}, state, {
-        currentMove: reshapeMove(
-          get(action.payload, 'service_member.orders.0.moves.0'),
-        ),
+        currentMove: reshapeMove(head(moves)),
+        canceledMove: find(moves, ['status', 'CANCELED']),
```
[the remark this answers is not in the document]
Comment (stangah) [reply src/scenes/Moves/ducks.js]: `find` will find the first item that matches the condition, and `moves` should come sorted from the server, so _ideally_ this does only find the most recent canceled move. You could totally make the argument though that I should just do the work to be extra sure here.

This is for the alert, my plan for now was to show an alert if any canceled move exists in history

File: src/scenes/Landing/MoveSummary.jsx (hunk 1 above)
Comment (stangah) [reply src/scenes/Landing/MoveSummary.jsx]: Is something like this more readable: https://github.com/transcom/mymove/commit/e483debce2d7b7fe98df19b8eca3dcc4aef33ee2

Comment (jim) [reply src/scenes/Landing/MoveSummary.jsx]: That version reads a lot better (for me at least).

File: src/scenes/Moves/ducks.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jim) [reply src/scenes/Moves/ducks.js]: You've thought about this, so I'm fine with leaving it as is for now and making changes in the future as we develop the notification system.

File: src/scenes/Landing/MoveSummary.test.jsx [hunk 3]
```diff
@@ -10,97 +16,149 @@ describe('MoveSummary', () => {
...
+          .dive()
+          .find('h2')
+          .html(),
+      ).toEqual('<h2>New move</h2>');
+    });
+  });
   describe('when a move is in submitted state', () => {
     it('renders submitted content', () => {
       const moveObj = { status: 'SUBMITTED' };
+      const canceledMoveObj = { status: 'CANCELED' };
```
Comment (amitch23) [inline src/scenes/Landing/MoveSummary.test.jsx]: Why pass in a canceled move here? Is it to test if a user has a past canceled move AND submitted move? Does this mean that as long as the user has canceled moves and even if they're starting another move, this alert shows?

File: src/scenes/Landing/MoveSummary.jsx [hunk 4]
```diff
@@ -12,342 +12,376 @@ import ppmSubmitted from './images/ppm-submitted.png';
...
+  } = props;
+  const status = get(move, 'status', 'DRAFT');
+  const StatusComponent = moveSummaryStatusComponents[status];
+  return (
+    <Fragment>
+      {canceledMove && (
+        <Alert type="info" heading="Your move was canceled">
+          Your move from {get(profile, 'current_station.name')} to{' '}
+          {get(orders, 'new_duty_station.name')} with the move locator ID{' '}
+          {get(canceledMove, 'locator')} was cancelled.
```
Comment (amitch23) [inline src/scenes/Landing/MoveSummary.jsx]: Non-british spelling  - "canceled"

File: src/scenes/Landing/MoveSummary.test.jsx (hunk 3 above)
Comment (stangah) [reply src/scenes/Landing/MoveSummary.test.jsx]: That's exactly right. Which, now that I think about, doesn't make sense? I don't know why I did it this way. Will fix.
